### PR TITLE
Add unit tests and documentation to `MultigridPatching2D`

### PR DESCRIPTION
**Pull request by dhpitt**
Adds in-line docstrings and unit tests for `neuralop.training.patching`.

### CODE REVIEW

File: neuralop/data/transforms/data_processors.py [hunk 1]
```diff
@@ -306,11 +307,11 @@ def __init__(
...
         levels : int
-            mg_patching level parameter for MultigridPatching2D
+            number of multi-grid patching levels to use
         padding_fraction : float
-            mg_padding_fraction parameter for MultigridPatching2D
-        stitching : float
-            mg_patching_stitching parameter for MultigridPatching2D
+            fraction by which to pad inputs in multigrid-patching
+        stitching : bool
+            whether to always perform stitching in multigrid-patching
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/data_processors.py:314]:
```suggestion
            whether to stitch back the output from the multi-grid patches 
```

File: neuralop/data/transforms/data_processors.py [hunk 2]
```diff
@@ -336,7 +338,7 @@ def __init__(
             self.in_normalizer = in_normalizer.to(self.device)
         if out_normalizer:
             self.out_normalizer = out_normalizer.to(self.device)
-        self.model = None
+        self.model = model
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/data_processors.py:341]: Do we use the model in the processor?

Comment (dhpitt) [reply neuralop/data/transforms/data_processors.py:341]: In distributed settings, the `MultigridPatching2D` takes the model and adds a hook to the parameters that averages gradients.

File: neuralop/data/transforms/data_processors.py [hunk 3]
```diff
@@ -384,7 +389,7 @@ def postprocess(self, out, data_dict):
             model output predictions
         """
         y = data_dict["y"]
-        out, y = self.patcher.unpatch(out, y)
+        out, y = self.patcher.unpatch(out, y, evaluation=not self.training)
```
[the remark this answers is not in the document]
Comment (dhpitt) [reply neuralop/data/transforms/data_processors.py:392]: this can be controlled by `stitching` in the patcher
